src/circuit/gadget/ecc/chip.rs

```diff
@@ -176,18 +213,115 @@ impl EccChip {
+            config.create_gate(meta);
+        }
+
+        // Create witness scalar_fixed gate that only applies to short scalars
+        {
+            let config: witness_scalar_fixed::short::Config = (&config).into();
+            config.create_gate(meta);
+        }
+
+        // Create fixed-base scalar mul gate that is used in both full-width
+        // and short multiplication.
+        {
+            let mul_fixed_config: mul_fixed::Config<{ constants::NUM_WINDOWS }> = (&config).into();
+            mul_fixed_config.create_gate_scalar(meta);
+        }
+
+        // Create gate that is only used in short fixed-base scalar mul.
+        {
+            let short_config: mul_fixed::short::Config<{ constants::NUM_WINDOWS_SHORT }> =
+                (&config).into();
+            short_config.create_gate(meta);
+        }
+
+        // Create gate that is only used in fixed-base mul using a base field element.
+        {
+            let base_field_config: mul_fixed::base_field_elem::Config = (&config).into();
+            base_field_config.create_gate(meta);
+        }
+
         config
     }
 }
 
+/// A base-field element used as the scalar in variable-base scalar multiplication.
+#[derive(Copy, Clone, Debug)]
+pub struct EccScalarVar(CellValue<pallas::Base>);
+impl std::ops::Deref for EccScalarVar {
+    type Target = CellValue<pallas::Base>;
+
+    fn deref(&self) -> &CellValue<pallas::Base> {
+        &self.0
+    }
+}
+
+/// A full-width scalar used for fixed-base scalar multiplication.
+/// This is decomposed into 85 3-bit windows in little-endian order,
+/// i.e. `windows` = [k_0, k_1, ..., k_84] (for a 255-bit scalar)
+/// where `scalar = k_0 + k_1 * (2^3) + ... + k_84 * (2^3)^84` and
+/// each `k_i` is in the range [0..2^3).
+#[derive(Clone, Debug)]
+pub struct EccScalarFixed {
+    value: Option<pallas::Scalar>,
+    windows: ArrayVec<CellValue<pallas::Base>, { constants::NUM_WINDOWS }>,
+}
+
+/// A signed short scalar used for fixed-base scalar multiplication.
+/// A short scalar must have magnitude in the range [0..2^64), with
+/// a sign of either 1 or -1.
+/// This is decomposed into 22 3-bit windows in little-endian order,
+/// i.e. `windows` = [k_0, k_1, ..., k_21] (for a 64-bit magnitude)
+/// where `scalar = k_0 + k_1 * (2^3) + ... + k_84 * (2^3)^84` and
+/// each `k_i` is in the range [0..2^3).
+/// k_21 must be a single bit, i.e. 0 or 1.
+#[derive(Clone, Debug)]
+pub struct EccScalarFixedShort {
+    magnitude: Option<pallas::Scalar>,
+    sign: CellValue<pallas::Base>,
+    windows: ArrayVec<CellValue<pallas::Base>, { constants::NUM_WINDOWS_SHORT }>,
+}
+
+/// A base field element used for fixed-base scalar multiplication.
+/// This is decomposed into 3-bit windows in little-endian order
+/// using a running sum `z`, where z_{i+1} = (z_i - a_i) / (2^3)
+/// for element α = a_0 + (2^3) a_1 + ... + (2^{3(n-1)}) a_{n-1}.
+/// Each `a_i` is in the range [0..2^3).
```

Comment:
This also applies to the comment above (which is confusing because that's the comment that will be read first).